e2e/cypress/integration/enterprise/ldap/ldap_group_sync_spec.js

```diff
@@ -0,0 +1,348 @@
+                cy.wait(TIMEOUTS.THREE_SEC); //eslint-disable-line cypress/no-unnecessary-waiting
+
+                // Make it public and save
+                // * Ensure it still showing the channel as private
+                cy.findByTestId('allow-all-toggle').should('has.have.text', 'Private').click();
+                cy.get('#saveSetting').should('be.enabled').click();
+                cy.get('#confirmModalButton').click();
+                cy.get('.admin-console__header', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').and('have.text', 'Mattermost Channels');
+
+                // Reload
+                cy.visit(`/admin_console/user_management/channels/${privateChannel.id}`);
+                cy.get('.admin-console__header', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').and('have.text', 'Channel Configuration');
+
+                // * Ensure it still showing the channel as private
+                cy.findByTestId('allow-all-toggle').should('has.have.text', 'Public');
+
+                // # Ensure the last message in the message says that it was converted to a public channel
+                cy.visit(`/${testTeam.name}/channels/${privateChannel.name}`);
+                cy.getLastPostId().then((id) => {
+                    // * The system message should contain 'This channel has been converted to a Public Channel and can be joined by any team member'
+                    cy.get(`#postMessageText_${id}`).should('contain', 'This channel has been converted to a Public Channel and can be joined by any team member');
+                });
+            });
+        });
+
+        it('MM-T2630 - Default channel cannot be toggled to private', () => {
+            cy.visit('/admin_console/user_management/channels');
+
+            // # Search for the channel town square
+            cy.get('.DataGrid_searchBar').within(() => {
+                cy.findByPlaceholderText('Search').should('be.visible').type('Town Square');
+            });
+            cy.wait(TIMEOUTS.FIVE_SEC); //eslint-disable-line cypress/no-unnecessary-waiting
+
+            cy.findAllByTestId('town-squareedit').then((elements) => {
+                elements[0].click();
+                cy.get('.admin-console__header', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').and('have.text', 'Channel Configuration');
+
+                // * Ensure the toggle to private/public is disabled
+                cy.findByTestId('allow-all-toggle-button').should('be.disabled');
+            });
+        });
+
+        it('MM-T2638 - Permalink from when public does not auto-join (non-system-admin) after converting to private', () => {
+            cy.apiLogin(testUser);
+
+            cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+
+            // # Post message to use
+            cy.postMessage('DONT YOU SEE I GOT EVERTYHING YOU NEED .... BABY BABY DONT YOU SEE SEE I GOT EVERYTHING YOU NEED NEED ... ;)');
```

Comment:
lol, sounds like you're inspired 😄 
```suggestion
            cy.postMessage('DONT YOU SEE I GOT EVERYTHING YOU NEED .... BABY BABY DONT YOU SEE SEE I GOT EVERYTHING YOU NEED NEED ... ;)');
```

Reply:
😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆 😆

Reply:
![dance](https://user-images.githubusercontent.com/17804942/91891662-6e039c00-ec5f-11ea-8705-38f7ae1c47da.gif)

Reply:
I see you discovered my hidden Easter egg 🤣😂😂😂☺️☺️🙂☺️